airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,142 @@
+                 timeout_seconds=60,
+                 retry_limit=3):
+        self.databricks_conn_id = databricks_conn_id
+        self.databricks_conn = self.get_connection(databricks_conn_id)
+        self.timeout_seconds = timeout_seconds
+        self.retry_limit = retry_limit
+
+    def _do_api_call(self, endpoint_info, api_parameters):
+        method, endpoint = endpoint_info
+        url = 'https://{host}/{endpoint}'.format(
+                host=self.databricks_conn.host,
+                endpoint=endpoint)
+        auth = (self.databricks_conn.login, self.databricks_conn.password)
+        if method == 'GET':
+            request_func = requests.get
+        elif method == 'POST':
+            request_func = requests.post
+        else:
+            raise AirflowException('Unexpected HTTP Method')
+
+        for _ in range(self.retry_limit):
+            attempt_num = _ + 1
+            try:
+                response = request_func(url,
+                                        json=api_parameters,
+                                        auth=auth,
+                                        headers=USER_AGENT_HEADER,
+                                        timeout=self.timeout_seconds,
+                                        verify=False)
+                if response.status_code == 404:
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
+                    raise AirflowException(response.content)
+                else:
+                    return response.json()
+            except Exception as e:
+                logging.error('Attempt {0} API Request to Databricks failed ' +
+                              'with reason: {1}'.format(attempt_num, e))
+        raise AirflowException('API requests to Databricks failed {} times. ' +
+                               'Giving up.'.format(self.retry_limit))
+
+    def submit_run(self,
+                   spark_jar_task=None,
+                   notebook_task=None,
+                   new_cluster=None,
+                   existing_cluster_id=None,
+                   libraries=[],
+                   run_name=None,
+                   timeout_seconds=0,
+                   **kwargs):
+        api_params = kwargs.copy()
+        if spark_jar_task is not None:
+            api_params['spark_jar_task'] = spark_jar_task
+        if notebook_task is not None:
+            api_params['notebook_task'] = notebook_task
+        if new_cluster is not None:
+            api_params['new_cluster'] = new_cluster
+        if existing_cluster_id is not None:
+            api_params['existing_cluster_id'] = existing_cluster_id
+        api_params['libraries'] = libraries
+        if run_name is not None:
+            api_params['run_name'] = run_name
+        api_params['timeout_seconds'] = timeout_seconds
+        response = self._do_api_call(SUBMIT_RUN_ENDPOINT, api_params)
+        return response['run_id']
+
+    def get_run_page_url(self, run_id):
```

Comment:
if we return the run_page_url with submit_run, then we don't need this API, right?